File: CC_Visual Basic/VisualBasic/BeginVBE/BeginVBE/BVBE Projects/GuessNumber/Form1.vb
Public Class Form1
  Dim TheNumber As Integer
  Dim MyGuess As Integer
  Dim MyRandom As New Random


 
  Private Sub btnPick_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnPick.Click
    If btnPick.Text = "Pick Number" Then
      'Get new number and set controls
      TheNumber = MyRandom.Next(101)
      txtMessage.Text = "I'm thinking of a number between 0 and 100"
      nudGuess.Value = 50
      nudGuess.Enabled = True
      btnCheck.Enabled = True
      btnPick.Text = "Show Answer"
    Else
      'Just show the answer and re-set controls
      txtMessage.Text = "The answer is" + Str(TheNumber)
      nudGuess.Value = TheNumber
      nudGuess.Enabled = False
      btnCheck.Enabled = False
      btnPick.Text = "Pick Number"
    End If
  End Sub

  Private Sub btnCheck_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCheck.Click
    'Guess is the updown control value
    MyGuess = nudGuess.Value
    If MyGuess = TheNumber Then
      'Correct guess
      txtMessage.Text = "That's it!!"
      nudGuess.Enabled = False
      btnCheck.Enabled = False
      btnPick.Text = "Pick Number"
    ElseIf MyGuess < TheNumber Then
      'Guess is too low
      txtMessage.Text = "Too low!"
    Else
      'Guess is too high
      txtMessage.Text = "Too high!"
    End If
  End Sub

  Private Sub btnExit_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnExit.Click
    Me.Close()
  End Sub
End Class

File: CC_C#/BVCS Projects/GuessNumber/Form1.cs
#region Using directives

using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Windows.Forms;

#endregion

namespace GuessNumber
{
    partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        int theNumber;
        int myGuess;
        Random myRandom = new Random();

        private void btnPick_Click(object sender, EventArgs e)
        {
            if (btnPick.Text == "Pick Number")
            {
                // Get new number and set controls
                theNumber = myRandom.Next(101);
                txtMessage.Text = "I'm thinking of a number between 0 and 100";
                nudGuess.Value = 50;
                nudGuess.Enabled = true;
                btnCheck.Enabled = true;
                btnPick.Text = "Show Answer";
            }
            else
            {
                // Just show the answer and re-set controls
                txtMessage.Text = "The answer is" + Convert.ToString(theNumber);
                nudGuess.Value = theNumber;
                nudGuess.Enabled = false;
                btnCheck.Enabled = false;
                btnPick.Text = "Pick Number";
            }
        }

        private void btnCheck_Click(object sender, EventArgs e)
        {
            // Guess is the updown control value
            myGuess = (int) nudGuess.Value;
            if (myGuess == theNumber)
            {
                // Correct guess
                txtMessage.Text = "That's it!!";
                nudGuess.Enabled = false;
                btnCheck.Enabled = false;
                btnPick.Text = "Pick Number";
            }
            else if (myGuess < theNumber)
            {
                // Guess is too low
                txtMessage.Text = "Too low!";
            }
            else
            {
                // Guess is too high
                txtMessage.Text = "Too high!";
            }
        }

        private void btnExit_Click(object sender, EventArgs e)
        {
            this.Close();
        }
    }
}